include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -75,13 +77,299 @@ namespace LinearAlgebra
                                "implemented for complex numbers"));
         return a;
       }
-    } // namespace internal
 
 
 
-    template <typename Number>
+      // Resize the underlying array on the host or on the device
+      template <typename Number, typename MemorySpace>
+      struct la_parallel_vector_templates_functions
+      {
```

Comment:
I would again put a `static_assert` inside these structs to make sure they are only used for ` Host` or `CUDA`.